department-of-veterans-affairs/caseflow
# MattT/APPEALS-8447: Adjust "Docket appeal" Modal for CSP

Resolves [APPEALS-8447](https://vajira.max.gov/browse/APPEALS-8447)

### Description
1. Allows for different organization names to be inserted into the body of the "Docket appeal" modal depending on which workflow the appeal has been through.
2. Makes the Submit button blue with a hover effect.
3. Adds "Optional" label to textarea field on the "Docket appeal" modal.

### Acceptance Criteria
- [ ] The following occurs when the "Docket appeal" task action is selected
     - [ ] A modal appears to confirm intent 
     - [ ] Modal Header: Docket appeal
     - [ ] Modal informational text: Please confirm that the document provided by CSP are available in VBMS before docketing this appeal.
         - [ ] Once you confirm, the appeal will be established. Please remember to send the docketing letter out to all parties and representatives.
         - [ ] Provide instructions and context for this action:
     - [ ] BVA Intake user can provide additional context for the action
         - [ ] Field type: optional free text field
     - [ ] Modal contains the following selectable options:
         - [ ] Cancel
            - [ ] User is returned to case details screen and selection is not retained
         - [ ] Confirm
            - [ ] Select to execute task action
- [ ] The following occurs when the "Docket appeal" task action is executed 
   - [ ] The predocket task is completed
   - [ ] The appeal appears on the Completed tab of BVA Intake organization queue
   - [ ] BVA Intake user is returned to the BVA Intake organization queue 
      - [ ] A message appears confirming the appeal has been docketed


### Testing Plan
1. Log into Caseflow as an Intake user (ex: `BVAISHAW`). You will need to intake three appeals with the specified issues:

   - [ ] VHA CAMO
      - [ ] Benefit Type: Veterans Health Administration
      - [ ] Issue Category: Any except for the caregiver ones
      
   - [ ] VHA CSP
      - [ ] Benefit Type: Veterans Health Administration
      - [ ] Issue Category: Any caregiver issue type
          
   - [ ] Education
      - [ ] Benefit Type: Education
      - [ ] Predocket Needed?: Yes
      - [ ] Issue Category: Any is fine
      
I'd recommend using three different veteran file numbers for an easier time validating the appeals' presences in the BVA Intake queue.
  
2. Navigate to the Case Details pages for all three appeals (`/queue/appeals/<UUID>`)

3. Select "Docket appeal" from the task actions dropdown. Ensure that the modals match the [designs](https://bipuiux.invisionapp.com/console/share/P2MB68RZ73F/933825336/play).

4. On at least one modal, submit it without entering any text into the text field. Submit the other two.

5. Navigate to BVA Intake's completed tab (`/organizations/bva-intake?tab=bvaIntakeCompletedTab`) All three appeals should be here now!
   


### User Facing Changes

 WORKFLOW|BEFORE|AFTER
 ---|---|---
 VHA CAMO|<img width="285" alt="pre-changes-docket" src="https://user-images.githubusercontent.com/99351305/185702741-bdcbafb8-4aec-43af-af38-6f5857299b43.PNG">|<img width="284" alt="post-changes-vha-docket" src="https://user-images.githubusercontent.com/99351305/185702851-e686ab56-1365-4ddc-a319-1fa05330d76e.PNG">
 VHA CSP|<img width="285" alt="pre-changes-docket" src="https://user-images.githubusercontent.com/99351305/185702741-bdcbafb8-4aec-43af-af38-6f5857299b43.PNG">|<img width="277" alt="post-changes-csp-docket" src="https://user-images.githubusercontent.com/99351305/185702827-12005e08-ed5c-489b-b929-8a02f45cfc71.PNG">
 Education (EMO & RPO)|<img width="285" alt="pre-changes-docket" src="https://user-images.githubusercontent.com/99351305/185702741-bdcbafb8-4aec-43af-af38-6f5857299b43.PNG">|<img width="290" alt="docket-appeal-edu-service" src="https://user-images.githubusercontent.com/99351305/187525666-58e5ac6b-3899-49fa-8a83-5cc416ec7a1d.PNG">

### Code Documentation Updates
